File: app/src/main/java/com/nicholasworkshop/tinklabstest/MainModule.java
package com.nicholasworkshop.tinklabstest;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by nicholas.wong on 2017/05/15.
 */
@Module
public class MainModule {

    private final Application application;

    MainModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Context context() {
        return application;
    }
}

File: app/src/main/kotlin/com/nicholasworkshop/tinklabstest/MainModule.kt
package com.nicholasworkshop.tinklabstest

import android.app.Application
import android.content.Context
import android.view.LayoutInflater

import com.bumptech.glide.Glide
import com.bumptech.glide.RequestManager

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
internal class MainModule(
        private val application: Application) {

    @Provides
    @Singleton
    fun context(): Context {
        return application
    }

    @Provides
    @Singleton
    fun requestManager(context: Context): RequestManager {
        return Glide.with(context)
    }

    @Provides
    @Singleton
    fun layoutInflater(context: Context): LayoutInflater {
        return LayoutInflater.from(context)
    }
}
